Plugins/GeoModel/src/GeoModelToDetVol.cpp

```diff
@@ -0,0 +1,182 @@
+        newTransform.rotate(
+            transform.rotation() *
+            Eigen::AngleAxisd(rotationAngle, Acts::Vector3::UnitX()));
+        volumes.emplace_back(Experimental::DetectorVolumeFactory::construct(
+            portalGenerator, context, name, newTransform, bounds,
+            Experimental::tryAllPortalsAndSurfaces()));
+      } else {
+        std::shared_ptr<TrapezoidVolumeBounds> bounds =
+            std::make_shared<TrapezoidVolumeBounds>(x2, x1, z, y1);
+        auto rotationAngle = M_PI;
+        Acts::Transform3 newTransform = Acts::Transform3::Identity();
+        newTransform.translate(transform.translation());
+        newTransform.rotate(
+            transform.rotation() *
+            Eigen::AngleAxisd(rotationAngle, Acts::Vector3::UnitY()) *
+            Eigen::AngleAxisd(rotationAngle, Acts::Vector3::UnitZ()));
+        volumes.emplace_back(Experimental::DetectorVolumeFactory::construct(
+            portalGenerator, context, name, newTransform, bounds,
+            Experimental::tryAllPortalsAndSurfaces()));
+      }
+    } else if (x1 == x2) {
+      if (y1 < y2) {
+        std::shared_ptr<TrapezoidVolumeBounds> bounds =
+            std::make_shared<TrapezoidVolumeBounds>(y1, y2, z, x1);
+        auto rotationAngle = M_PI / 2;
+        Acts::Transform3 newTransform = Acts::Transform3::Identity();
+        newTransform.translate(transform.translation());
+        newTransform.rotate(
+            transform.rotation() *
+            Eigen::AngleAxisd(rotationAngle, Acts::Vector3::UnitZ()) *
+            Eigen::AngleAxisd(rotationAngle, Acts::Vector3::UnitX()));
+        volumes.emplace_back(Experimental::DetectorVolumeFactory::construct(
+            portalGenerator, context, name, newTransform, bounds,
+            Experimental::tryAllPortalsAndSurfaces()));
+      } else {
+        std::shared_ptr<TrapezoidVolumeBounds> bounds =
+            std::make_shared<TrapezoidVolumeBounds>(y2, y1, z, x1);
+        auto rotationAngle = M_PI;
+        Acts::Transform3 newTransform = Acts::Transform3::Identity();
+        newTransform.translate(transform.translation());
+        newTransform.rotate(
+            transform.rotation() *
+            Eigen::AngleAxisd(rotationAngle, Acts::Vector3::UnitX()) *
+            Eigen::AngleAxisd(rotationAngle / 2, Acts::Vector3::UnitZ()) *
+            Eigen::AngleAxisd(rotationAngle / 2, Acts::Vector3::UnitX()));
+        volumes.emplace_back(Experimental::DetectorVolumeFactory::construct(
+            portalGenerator, context, name, newTransform, bounds,
+            Experimental::tryAllPortalsAndSurfaces()));
+      }
+    } else {
+      throw std::runtime_error("FATAL: Translating GeoTrd to ACTS failed");
+    }
+  } else if (shape.typeID() == GeoShapeUnion::getClassTypeID()) {
+    const GeoShapeUnion& unionShape = static_cast<const GeoShapeUnion&>(shape);
+    const GeoShape* shapeA = unionShape.getOpA();
+    const GeoShape* shapeB = unionShape.getOpB();
```

Comment:
The surronding box doesn't work?

Reply:
No it doesn't work for the union. I also think this will be a better solution anyway. I think breaking up the union will give an overall smaller bounding box since we are breaking it up into components and this will lead to less clashes between volumes.